src/communication/network_interface_cl.c

```diff
@@ -10560,3 +10561,160 @@ loaddb_update_stats ()
+  int num_tables = 0;
+
+  char table[SM_MAX_IDENTIFIER_LENGTH];
+
+  OR_ALIGNED_BUF (OR_INT_SIZE + OR_INT_SIZE) a_reply;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  char *area;
+  int area_size;
+
+  num_tables = da_size (class_list);
+
+  /* request : num tables | table name list | user | start_time | end_time */
+  request_size = or_packed_string_length (user, NULL) + OR_INT64_SIZE + OR_INT64_SIZE;
+
+  for (int i = 0; i < num_tables; i++)
+    {
+      if (da_get (class_list, i, table) != NO_ERROR)
+	{
+	  return ER_FAILED;
+	}
+      request_size += or_packed_string_length (table, NULL);
+    }
+
+  request = (char *) malloc (request_size + MAX_ALIGNMENT);
+  if (request == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, (size_t) request_size);
+      return ER_FAILED;
+    }
+
+  ptr = start_ptr = PTR_ALIGN (request, MAX_ALIGNMENT);
+
+  ptr = or_pack_int (ptr, num_tables);
+  for (int i = 0; i < num_tables; i++)
+    {
+      if (da_get (class_list, i, table) != NO_ERROR)
+	{
+	  free_and_init (request);
+	  return ER_FAILED;
+	}
+      ptr = or_pack_string (ptr, table);
+    }
+  ptr = or_pack_string (ptr, user);
+  ptr = or_pack_int64 (ptr, start_time);
+  ptr = or_pack_int64 (ptr, end_time);
+
+  request_size = ptr - start_ptr;
+
+  req_error =
+    net_client_request2 (NET_SERVER_FLASHBACK_GET_SUMMARY, start_ptr, request_size, reply,
+			 OR_ALIGNED_BUF_SIZE (a_reply), NULL, 0, &area, &area_size);
+
+  if (!req_error)
+    {
+      ptr = or_unpack_int (reply, &area_size);
+      ptr = or_unpack_int (ptr, &rep_error);
+      if (area_size > 0)
+	{
+	  ptr = area;
+	  /* area : OID list | summary info list  */
+	  for (int i = 0; i < num_tables; i++)
+	    {
+	      ptr = or_unpack_oid (ptr, &(*oid_list)[i]);
+	    }
+	  /* get summary info */
+	}
+
+      free_and_init (area);
+    }
+
+  free_and_init (request);
+
+  return rep_error;
+#endif // CS_MODE
+  return ER_NOT_IN_STANDALONE;
+}
+
+int
+flashback_get_loginfo (int trid, char *user, OID * classlist, int num_table, LOG_LSA * start_lsa, LOG_LSA * end_lsa,
+		       int *num_item, bool forward, void *info_list)
+{
+#if defined(CS_MODE)
+  int req_error = ER_FAILED;
+  int rep_error = ER_FAILED;
+  int request_size = 0;
+  char *request = NULL, *ptr, *start_ptr;
+
+  OR_ALIGNED_BUF (OR_INT_SIZE + OR_INT_SIZE) a_reply;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  char *area;
+  int area_size;
+
+  /* request with : tranid, user, num class, classlist, start_lsa, end_lsa, num_item, forward/backward */
+
+  request_size =
+    OR_INT_SIZE + or_packed_string_length (user,
+					   NULL) + OR_OID_SIZE * num_table + OR_LOG_LSA_ALIGNED_SIZE * 2 + OR_INT_SIZE +
+    OR_INT_SIZE;
+
+  request = (char *) malloc (request_size + MAX_ALIGNMENT);
+  if (request == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, (size_t) request_size);
+      return ER_FAILED;
```

Comment:
```suggestion
      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, (size_t) request_size);
      return ER_OUT_OF_VIRTUAL_MEMORY;
```
It doesn't need to match?

Reply:
yes, I agree. I will change it